### PR TITLE
use istioctl binary from downloaded tarball instead of downloading from istioctl_url

### PR DESCRIPTION
**Pull request by rkpagadala**
using istioctl from the downloaded tarball allows us not to store istioctl in the release buckets.

Needed to fix https://github.com/istio/istio/issues/2904

### CODE REVIEW

**Comment by hklai** [comment]
/lgtm
/approve

It is a good thing anyway as we want to test the locally built istioctl.


However, why are these two scripts so similar? Can we refactor them and just parameterize the differences?

File: prow/e2e-suite.sh [hunk 1]
```diff
@@ -67,6 +66,9 @@ trap cleanup EXIT
 # use uploaded yaml artifacts rather than the ones generated locally
 DAILY_BUILD=istio-$(echo ${ISTIOCTL_URL} | cut -d '/' -f 6)
 LINUX_DIST_URL=${ISTIOCTL_URL/istioctl/${DAILY_BUILD}-linux.tar.gz}
+#disable ISTIOCTL_URL
+export ISTIOCTL_URL=""
```
Comment (chxchx) [inline prow/e2e-suite.sh]: perhaps `unset ISTIOCTL_URL` is a better practice?

**Comment by sebastienvas** [comment]
I forgot to update the  cluster-wide-auth file that you just updated. Can you move your change in e2e-suite.sh and just call e2e-suite.sh --cluster_wide --enable_auth after set-x

File: prow/daily-e2e-cluster_wide-auth.sh [hunk 2]
```diff
@@ -61,6 +60,9 @@ trap "cp -a /tmp/istio* ${GOPATH}/src/istio.io/istio/_artifacts" EXIT
 # use uploaded yaml artifacts rather than the ones generated locally
 DAILY_BUILD=istio-$(echo ${ISTIOCTL_URL} | cut -d '/' -f 6)
 LINUX_DIST_URL=${ISTIOCTL_URL/istioctl/${DAILY_BUILD}-linux.tar.gz}
+#disable ISTIOCTL_URL
+export ISTIOCTL_URL=""
```
Comment (chxchx) [inline prow/daily-e2e-cluster_wide-auth.sh]: same here

**Comment by sebastienvas** [comment]
or I approve this one and will fix it

**Comment by chxchx** [comment]
> @hklai: we want to test the locally built istioctl.

That is quite the opposite of what this PR is trying to do. In pre/postsubmits, we test with locally built artifacts such as istioctl and deployment yamls. But for release, we want to test the exact artifacts we are shipping, which means we should not use any artifacts that are locally built but rather replace them with exactly what the daily build produces upstream. 

As for the reason for two sets of scripts, my guess is that prior to mason preprovisioning, namespace-wide tests share the same cluster while cluster wide creates a cluster for itself, so the way to set up and trigger them is different. But given mason, merging the two sets of scripts should be doable. @sebastienvas probably knows more on this topic and please chime in if my understanding is biased.

**Comment by rkpagadala** [comment]
PTAL address comments. Agree with @chxchx that we want to test downloaded istioctl from tarball and not download again with a different url.

**Comment by hklai** [comment]
@chxchx Thanks.  Maybe I don't fully understand it. The new code uses "istioctl "${GOPATH}/src/istio.io/istio/${DAILY_BUILD}/bin/istioctl"". Is it using the istioctl from the release or the locally built one?

@rkpagadala if one script simply calls the other, why do we need to scripts at all? Can we just fix the caller of daily-e2e-cluster_wide-auth.sh to call e2e_suite.sh?

**Comment by rkpagadala** [comment]
@hklai  istioctl is from the release 
```
wget $LINUX_DIST_URL
tar -xzf ${DAILY_BUILD}-linux.tar.gz
cp -R ${DAILY_BUILD}/install/* install/
```

the pwd is ${GOPATH}/src/istio.io/istio/ at that point

the need for script is because prow needs to call scripts. Even if we can fix it, it is beyond the scope of this PR

**Pull request merged**